src/baldr/edgeinfo.cc

```diff
@@ -415,24 +467,43 @@ int8_t EdgeInfo::layer() const {
   return static_cast<int8_t>(value.front());
 }
 
-std::string EdgeInfo::level() const {
+bool EdgeInfo::includes_level(float lvl) {
   const auto& tags = GetTags();
-  auto itr = tags.find(TaggedValue::kLevel);
+  auto itr = tags.find(TaggedValue::kLevels);
   if (itr == tags.end()) {
-    return "";
+    return false;
   }
-  const std::string& value = itr->second;
-  return value;
+  try {
+    auto decoded = std::get<0>(decode_levels(itr->second));
+    // TODO: use binary search
```

Comment:
if we construct a vector of pairs instead, we could use `std::lower_bound` or `upper_bound`, which would be using binary search.